fixtures/node-app-pages/tests/index.test.ts

```diff
@@ -1,36 +1,46 @@
 import { fork } from "child_process";
 import path from "path";
 import { fetch } from "undici";
-import type { ChildProcess } from "child_process";
 
 describe("Pages Dev", () => {
-	let wranglerProcess: ChildProcess;
-	let ip: string;
-	let port: number;
-	let resolveReadyPromise: (value: unknown) => void;
-	const readyPromise = new Promise((resolve) => {
-		resolveReadyPromise = resolve;
-	});
+	it.concurrent(
+		"should work with `--node-compat` when running code requiring polyfills",
+		async () => {
+			const { ip, port, exit } = await runWranglerPagesDev(
+				"public",
+				"--node-compat",
+				"--port=0"
+			);
+			const response = await fetch(`http://${ip}:${port}/stripe`);
 
-	beforeAll(() => {
-		wranglerProcess = fork(
-			path.join("..", "..", "packages", "wrangler", "bin", "wrangler.js"),
-			["pages", "dev", "public", "--node-compat", "--port=0"],
-			{
-				stdio: ["inherit", "inherit", "inherit", "ipc"],
-				cwd: path.resolve(__dirname, ".."),
-			}
-		).on("message", (message) => {
-			const parsedMessage = JSON.parse(message.toString());
-			ip = parsedMessage.ip;
-			port = parsedMessage.port;
-			resolveReadyPromise(undefined);
-		});
-	});
+			await expect(response.text()).resolves.toContain(
+				`"PATH":"path/to/some-file","STRIPE_OBJECT"`
+			);
 
-	afterAll(async () => {
-		await readyPromise;
-		await new Promise((resolve, reject) => {
+			await exit();
+		},
+		10000
+	);
+});
+
+async function runWranglerPagesDev(publicPath: string, ...options: string[]) {
```

Comment:
I feel like this function might be clearer if it was entirely wrapped in a Promise constructor callback, rather than reassigning the resolve function—I could be wrong though?